middleware/cors/cors_test.go

```diff
@@ -647,3 +674,51 @@ func Test_CORS_AllowCredentials(t *testing.T) {
 		})
 	}
 }
+
+func Benchmark_CORS_NewHandler(b *testing.B) {
+	app := fiber.New()
+	app.Use(New(Config{
+		AllowOrigins:     "http://localhost,http://example.com",
+		AllowMethods:     "GET,POST,PUT,DELETE",
+		AllowHeaders:     "Origin,Content-Type,Accept",
+		AllowCredentials: true,
+		MaxAge:           600,
+	}))
+
+	req := httptest.NewRequest(fiber.MethodGet, "/", nil)
+	req.Header.Set(fiber.HeaderOrigin, "http://localhost")
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		resp, err := app.Test(req)
+		if err != nil {
+			b.Fatalf("Failed to perform request: %v", err)
+		}
+		_ = resp
+	}
+}
+
+func Benchmark_CORS_NewHandlerPreflight(b *testing.B) {
+	app := fiber.New()
+	app.Use(New(Config{
+		AllowOrigins:     "http://localhost,http://example.com",
+		AllowMethods:     "GET,POST,PUT,DELETE",
+		AllowHeaders:     "Origin,Content-Type,Accept",
+		AllowCredentials: true,
+		MaxAge:           600,
+	}))
+
+	req := httptest.NewRequest(fiber.MethodOptions, "/", nil)
+	req.Header.Set(fiber.HeaderOrigin, "http://localhost")
+	req.Header.Set(fiber.HeaderAccessControlRequestMethod, fiber.MethodPost)
+	req.Header.Set(fiber.HeaderAccessControlRequestHeaders, "Origin,Content-Type,Accept")
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		resp, err := app.Test(req)
```

Comment:
app.Test() is not intended for benchmarks. It converts your request adding a ton of overhead.